packages/connectivity/connectivity_web/test/lib/main.dart

```diff
@@ -0,0 +1,93 @@
+import 'package:e2e/e2e.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:connectivity_platform_interface/connectivity_platform_interface.dart';
+import 'package:connectivity_web/connectivity_web.dart';
+
+import 'src/connectivity_mocks.dart';
+
+void main() {
+  E2EWidgetsFlutterBinding.ensureInitialized();
+
+  group('checkConnectivity', () {
+    void testCheckConnectivity({
+      String type,
+      String effectiveType,
+      num downlink = 10,
+      num rtt = 50,
+      ConnectivityResult expected,
+    }) {
+      MockNetworkInformation connection = MockNetworkInformation(
+          type: type,
+          effectiveType: effectiveType,
+          downlink: downlink,
+          rtt: rtt);
+      ConnectivityPlugin plugin = ConnectivityPlugin.withConnection(connection);
+      expect(plugin.checkConnectivity(), completion(equals(expected)));
+    }
+
+    group('in Chrome', () {
```

Comment:
suggestion: we can add a browser based skip, if we want to show if something runs on chrome but does not run in any other browser. In CI we can run it against two different drivers? (we can also do it later when we are adding this to CI)

Reply:
Is this done with the `@TestOn('chrome')` annotation, or do I need to split this test in different files per-browser?

Reply:
you can also use skip: https://github.com/flutter/engine/blob/master/lib/web_ui/test/engine/semantics/semantics_test.dart#L168

Reply:
Interesting! Right now I'm mocking the behavior of other browsers, but that can get obsolete. It'd be great to be able to run these tests in multiple browsers and get the actual API doing its thing.

Reply:
Yes we can run it on multiple browsers when you are adding them to CI.

We can change the argument for browser-name, and choose another browsers when we are starting a drive test. 

https://github.com/flutter/flutter/blob/master/.cirrus.yml#L288